parsec/serde/fields.py

```diff
@@ -69,84 +69,108 @@
 )
 
 
-class BaseEnumField(Field):
-    ENUM: Enum
+class WithStrAttribute(Protocol):
+    def __init__(self, value: str) -> None:
+        ...
+
+    @property
+    def str(self) -> str:
+        ...
+
+
+RustEnumSelf = TypeVar("RustEnumSelf", bound="RustEnum")
+
+
+class RustEnum(Protocol):
+    @classmethod
+    def from_str(cls: Type[RustEnumSelf], value: str) -> RustEnumSelf:
+        ...
+
+    @property
+    def str(self) -> str:
+        ...
+
+
+class UUIDProtocol(Protocol):
+    @classmethod
+    def __init__(self, value: _UUID) -> None:
+        ...
+
+    @property
+    def uuid(self) -> _UUID:
+        ...
+
 
-    def _serialize(self, value: Any, attr: str, obj: object) -> str | None:
+T = TypeVar("T")
+E = TypeVar("E", bound=Enum)
+S = TypeVar("S", bound=WithStrAttribute)
+R = TypeVar("R", bound=RustEnum)
+U = TypeVar("U", bound=UUIDProtocol)
```

Comment:
```suggestion
T = TypeVar("T")
E = TypeVar("E", bound=Enum)

S = TypeVar("S", bound=WithStrAttribute)
U = TypeVar("U", bound=UUIDProtocol)
R = TypeVar("R", bound=RustEnum)
```
? (it's just a pun :stuck_out_tongue:)